docs/source/guide/configuration.rst

```diff
@@ -48,9 +50,83 @@ To set these configuration options, create a ``Config`` object with the options
 
     client = boto3.client('kinesis', config=my_config)
 
+
+Using proxies
+~~~~~~~~~~~~~
+Boto3 supports using proxies as intermediaries between your code and AWS. Proxies may provide any number of functions, from filtering to security and firewalls to privacy assurance.
+
+
+.. _specify_proxies:
+
+Specifying proxy servers
+''''''''''''''''''''''''
+
+You can specify proxy servers to be used when communicating between specific endpoints, or when
+using specific protocols. The ``proxies`` option in the ``Config`` object is a dictionary that maps
+the protocol name or endpoint URL to the address of a proxy server.
+
+.. code-block:: python
+
+    import boto3
+    from botocore.config import Config
+
+    proxy_list = {
+        'http': 'proxy.example.com:6502',
+        'http://mysite.org': 'proxy.example.org:2010'
+    }
+
+    my_config = Config(
+        'region_name': 'us-east-2',
+        'signature_version': 'v4',
+        'proxies': proxy_list
+    }
+
+    client = boto3.client('kinesis', config=my_config)
+
+
+In the above example, a proxy list is set up to use ``proxy.example.com``, port 6502 as the proxy for all HTTP requests by default. For requests to ``http://mysite.org``, port 2010 on ``proxy.example.org`` is used instead.
+
+
+.. _configure_proxies:
+
+Configuring proxies
+'''''''''''''''''''
+You can configure proxy usage with the ``proxies-config`` option, which is a dictionary that specifies the values of several proxy options by name. The valid keys for this dictionary are:
+
+* ``proxy_ca_bundle`` (string) - The pathname of a custom certificate bundle to use when establishing TLS/SSL connections using a proxy.
+* ``proxy_client_cert`` (string or tuple) - Specifies the certificate to use for proxy TLS client authentication. When the value is a string, it's the pathname of a certificate file. If it's a tuple it must be two comma-separated strings. The first is used as the pathname of the client certificate, and the second is the path to the certificate's key.
+* ``proxy_use_forwarding_for_https`` (Boolean) - When true, HTTPS requests which specify absolute URIs will be forwarded using the proxy. Only set this to true for highly trusted or corporate proxies; otherwise, the proxy can become a man-in-the-middle attack vector.
+
+.. code-block:: python
+
+    import boto3
+    from botocore.config import Config
+
+    proxy_list = {
+        'http': 'proxy.example.com:6502'
+    }
+
+    my_config = Config(
+        'region_name': 'us-east-2',
+        'signature_version': 'v4',
+        'proxies': proxy_list,
+        'proxy_config': {
+            'proxy_client_cert': '/path/of/certificate',
+            'proxy_use_forwarding_for_https': False
```

Comment:
This is `False` by default and I don't know if we need users to be touching it in most cases. It's for very specific setups, so it would be best not to use it in the common case.